src/goGenerateTests.ts

```diff
@@ -193,12 +186,9 @@ function generateTests(conf: Config, goConfig: vscode.WorkspaceConfiguration): T
 	});
 }
 
-function getFunctions(doc: vscode.TextDocument): Thenable<vscode.SymbolInformation[]> {
-	let documentSymbolProvider = new GoDocumentSymbolProvider();
-	return documentSymbolProvider
-		.provideDocumentSymbols(doc, null)
-		.then(symbols =>
-			symbols.filter(sym =>
-				sym.kind === vscode.SymbolKind.Function)
-		);
+async function getFunctions(doc: vscode.TextDocument): Promise<vscode.DocumentSymbol[]> {
+	const documentSymbolProvider = new GoDocumentSymbolProvider();
+	const symbols = await documentSymbolProvider.provideDocumentSymbols(doc, null);
+	const symbols_1 = symbols[0].children;
+	return symbols_1.filter(sym => sym.kind === vscode.SymbolKind.Function);
```

Comment:
Pardon my Go ignorance here...
We are picking up on only the functions that are the first level children of the `package` as per the data returned from the `go-outline` tool.
Does Go have any concept where there could functions at a deeper level?

Reply:
@ramya-rao-a plain(named) functions can only be declared on the top level (last time I checked, I doubt anything changed in 1.12 ). 

You can have anonymous functions declared and assigned to variables though
```go
funcVar := func(arg string) {}
```
but this is not exactly the same thing.